tools/nightly.py

```diff
@@ -238,21 +246,35 @@ def __init__(
         self.pip_source = pip_source
         self.base_executable = Path(base_executable or sys.executable).absolute()
         self._executable: Path | None = None
-        self._env = {"PIP_EXTRA_INDEX_URL": self.pip_source.index_url}
+        self._bindir: Path | None = None
+        self._env = {
+            "PIP_EXTRA_INDEX_URL": self.pip_source.index_url,
+            "UV_INDEX": self.pip_source.index_url,
```

Comment:
Not adding `UV_TORCH_BACKEND=os.path.basename(self.pip_source.index_url)` here for nightly index URL.

See also:

- https://github.com/astral-sh/uv/issues/14091#issuecomment-2988544248